### ISSUE TITLE
How to make inference from the trained model?

### ISSUE THREAD

**Issue by geo47** [shenweichen/DeepCTR-Torch#198]
Hello,

Once a model is trained, how to save the checkpoint and reload for inferencing.
I didn't find any inference API in this library.

Does it support inferencing for the prediction task, like give user_id, movie_id, it will predict the rating?

Thanks.

**Comment by zanshuxun** [comment]
1. save and reload:
https://deepctr-torch.readthedocs.io/en/latest/FAQ.html#save-or-load-weights-models

2. inferencing
https://github.com/shenweichen/DeepCTR-Torch/blob/b4d8181e86c2165722fa9331bc16185832596232/examples/run_classification_criteo.py#L63

**Comment by geo47** [comment]
Hi @zanshuxun 

Thanks for the answer, 
While I was working on my custom dataset, I found out difficult to add node features information (like age, gender, genre, etc) for generating embedding.

Do you also provide any colab example for testing the library and customize it according to dataset.

Many thanks.

**Comment by zanshuxun** [comment]
Hi Muzamil,

1. Adding features is provided in our examples, please refer to https://github.com/shenweichen/DeepCTR-Torch/blob/master/examples/run_multivalue_movielens.py, which contains age, gender, genre, etc

2. For your customized dataset, just modify the `test_model_input` according to your feature columns to test the model:
https://github.com/shenweichen/DeepCTR-Torch/blob/b4d8181e86c2165722fa9331bc16185832596232/examples/run_classification_criteo.py#L63-L66

**Comment by geo47** [comment]
Hi @zanshuxun ,

Thanks for very informative answer.
Just curious if I could utilize `GraphSAGE embedding` to train the model.

For example:
I generate GraphSAGE embedding from `Neo4j database` with the node` User (age, gender, occupation), Movie(title, genre)` and an edge` User - [rate:4.0] -> Movie `
The output I get is a fixed sized vector for each user and movie
U1: [....]
M1: [...]
Then I pass those embedding to the neural network model for training with the label `rating`.
Can I achieve this with model supported by the DeepCTR library, If yes then what would be best approach.
Thanks in advance.

**Comment by zanshuxun** [comment]
So now you have pretrained user embeddings and move embeddings, then you want to feed these embeddings into DeepCTR models, right?

I would use these embeddings as pretrained weights of user/movie embedding layers, then finetune the model. Using pretrained weights is supported in the tensorflow version [DeepCTR ](https://github.com/shenweichen/DeepCTR). Details can be found in:
https://deepctr-doc.readthedocs.io/en/latest/FAQ.html#how-to-use-pretrained-weights-to-initialize-embedding-weights-and-frozen-embedding-weights

**Comment by treper** [comment]
@geo47 Hi, does using pre trained embedding and frozen them work in inference? I think embeddings vary with the model training them when it got retrained.  Unless you never change the embedding any more.
